From 1c7541cbb3bbe08841ef063abc01147d69fa8224 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <jak@debian.org>
Date: Wed, 7 Nov 2012 16:38:26 +0100
Subject: [PATCH] dh_autoreconf_clean: Correctly handle spaces in file names
 (Closes: #692317)

We split the checksums and filenames by simply using split, causing
us to ignore any part of the filename after a space. Limit the split
to split the line into two fields only.

This bug does not affect most projects, even if they have spaces
in their file names. It only affects projects where two files
in the same directory have a common prefix followed by a space,
and one of the files is modified (or the order they are visited
in changed).
---
 dh_autoreconf_clean |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dh_autoreconf_clean b/dh_autoreconf_clean
index 8d399d9..d05b9a5 100755
--- a/dh_autoreconf_clean
+++ b/dh_autoreconf_clean
@@ -44,7 +44,7 @@ open(FILE, 'debian/autoreconf.before') or die($!);
 while(<FILE>) {
     chomp($_);
     # The delimiter here is a comma
-    my ($checksum, $filename) = split;
+    my ($checksum, $filename) = split(' ', $_, 2);
     # Put the key => value pair in the hash
     $file{$filename} = $checksum;
 }
@@ -56,7 +56,7 @@ open(FILE, 'debian/autoreconf.after') or die($!);
 my $ltcheck = "";
 while(<FILE>) {
     chomp($_);
-    my ($checksum, $filename) = split;
+    my ($checksum, $filename) = split(' ', $_, 2);
 
     if ($filename eq "/usr/share/libtool/config/ltmain.sh") {
         $ltcheck = $checksum;
-- 
1.7.10.4

